src/runtime/tuple_parser.c

```diff
@@ -35,15 +35,12 @@ static charset charset_from_string(heap h, char *elements)
     return t;
 }
 
-// kind of a special case y combinator, try to fix the
-// allocation issues
-// XXX - can recycle closure?
+// kind of a special case y combinator
 closure_function(2, 1, parser, selfinate,
```

Comment:
oh good. do we still really need selfinate, maybe you're just trying to minimize disruption here

Reply:
Ok, I wasn't sure how to simplify this, but on further inspection - we shouldn't need the self_parser type at all, right? So maybe there's no need for combinate or selfinate if a parser can readily refer to itself. I'll give it a shot.

Reply:
I think so..but this is just based on my memory of saying 'well, we don't have self, so I guess we're going to have to do this ugly thing'...so I'm just assuming that since now we do, we don't